File: 2-year/Java/Lab1/src/main/java/com/Starygin/Presentation/Scenarios/WithdrawMoneyScenario.java
package com.Starygin.Presentation.Scenarios;

import com.Starygin.Application.Contracts.UserService;
import com.Starygin.Presentation.Scenario;

import java.io.IOException;
import java.util.Scanner;

public class WithdrawMoneyScenario implements Scenario {
    private final UserService userService;

    public WithdrawMoneyScenario(UserService userService) {
        this.userService = userService;
    }

    @Override
    public String getName() {
        return "Withdraw Money";
    }

    @Override
    public void run() {
        System.out.println("Enter account id: ");
        Scanner scanner = new Scanner(System.in);

        try {
            long accountId = scanner.nextLong();
            System.out.println("Enter money amount: ");
            double amount = scanner.nextDouble();

            userService.withdrawMoney(accountId, amount);

            System.out.println("Successfully withdrawn money");

        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: 2-year/OOP-CSharp/src/Lab5/Presentation/Lab5.Presentation.Console/Scenarios/WithdrawMoney/WithdrawMoneyScenario.cs
using Lab5.Application.Contracts.Users;
using Lab5.Application.Contracts.Users.OperationsResults;
using Spectre.Console;

namespace Lab5.Presentation.Console.Scenarios.WithdrawMoney;

public class WithdrawMoneyScenario : IScenario
{
    private readonly IUserService _userService;

    public WithdrawMoneyScenario(IUserService userService)
    {
        _userService = userService;
    }

    public string Name => "Withdraw money";

    public void Run()
    {
        long accountId = AnsiConsole.Ask<long>("Enter your account id: ");
        decimal amount = AnsiConsole.Ask<decimal>("Enter money amount: ");

        WithdrawMoneyResult result = _userService.WithdrawMoney(accountId, amount);

        string message = result switch
        {
            WithdrawMoneyResult.Success => "Success!",
            WithdrawMoneyResult.NotEnoughMoney => "Not enough money!",
            WithdrawMoneyResult.AccountNotFound => "Account not found",
            WithdrawMoneyResult.NotLoginYet => "Please login to continue",
            WithdrawMoneyResult.NegativeAmountOfMoney => "Amount of money cannot be negative",
            _ => throw new ArgumentOutOfRangeException(nameof(result)),
        };

        AnsiConsole.WriteLine(message);
        AnsiConsole.Ask<string>("Ok");
    }
}